lib/registry/client.go

```diff
@@ -362,13 +363,33 @@ func (c DockerRegistryClient) pullLayerHelper(
 
 // PushLayer pushes the image layer to the registry.
 func (c DockerRegistryClient) PushLayer(layerDigest image.Digest) error {
-	return c.pushLayerHelper(layerDigest, false)
+	return c.pushLayerWithBackoff(layerDigest, false)
 }
 
 // PushImageConfig pushes image config blob to the registry.
 // Same as PushLayer, with slightly different log message.
 func (c DockerRegistryClient) PushImageConfig(layerDigest image.Digest) error {
-	return c.pushLayerHelper(layerDigest, true)
+	return c.pushLayerWithBackoff(layerDigest, true)
+}
+
+func (c DockerRegistryClient) pushLayerWithBackoff(layerDigest image.Digest, isConfig bool) error {
+	multiError := utils.NewMultiErrors()
+	b := c.config.backoff()
+	for {
+		err := c.pushLayerHelper(layerDigest, isConfig)
+		// TODO: break on non-retryable errors.
```

Comment:
Are you planning to update this in the current PR?

Reply:
will have a follow-up PR for this as it is going to touch more parts in the registry client.